orttraining/orttraining/models/runner/training_util.h

```diff
@@ -54,6 +56,9 @@ class DataSet {
 
   void RandomShuffle();
 
+  void DataSet::GetSequenceMetrics(size_t batch_size, const std::map<std::string, std::pair<std::string, size_t>>& metrics_map,
```

Comment:
don't need the `DataSet::` here. some documentation would be useful, especially about how to use metrics_map and perf_properties

Reply:
can this function be const? also is this logic kind of specialized to BERT? would it make sense to just provide the relevant accessors in DataSet and compute it in the BERT-specific code?

Reply:
it causes build error

Reply:
Other model like GPT2 will use it too. Trying to make this function more generic. However, it's specific for getting information from input tensor shape.